test/ocsp/helper/helper.go

```diff
@@ -16,12 +16,45 @@ import (
 	"golang.org/x/crypto/ocsp"
 )
 
-var method = flag.String("method", "GET", "Method to use for fetching OCSP")
-var urlOverride = flag.String("url", "", "URL of OCSP responder to override")
-var hostOverride = flag.String("host", "", "Host header to override in HTTP request")
-var tooSoon = flag.Int("too-soon", 76, "If NextUpdate is fewer than this many hours in future, warn.")
-var ignoreExpiredCerts = flag.Bool("ignore-expired-certs", false, "If a cert is expired, don't bother requesting OCSP.")
-var expectStatus = flag.Int("expect-status", 0, "Expect response to have this numeric status (0=good, 1=revoked)")
+// CheckOCSPConfig contains fields which control various behaviors of the
+// checker's behavior.
+type CheckOCSPConfig struct {
+	method             string
+	urlOverride        string
+	hostOverride       string
+	tooSoon            int
+	ignoreExpiredCerts bool
+	expectStatus       int
+	expectReason       int
+}
+
+// WithExpectStatus returns a new CheckOCSPConfig with the given expectStatus,
+// and all other fields the same as the receiver.
+func (template CheckOCSPConfig) WithExpectStatus(status int) CheckOCSPConfig {
+	ret := template
+	ret.expectStatus = status
+	return ret
+}
+
+// DefaultConfig is the default configuration, as set by the command line flags
+// (or their default values, when not provided).
+var DefaultConfig CheckOCSPConfig
+
+func init() {
+	flag.StringVar(&DefaultConfig.method, "method", "GET", "Method to use for fetching OCSP")
+	flag.StringVar(&DefaultConfig.urlOverride, "url", "", "URL of OCSP responder to override")
+	flag.StringVar(&DefaultConfig.hostOverride, "host", "", "Host header to override in HTTP request")
+	flag.IntVar(&DefaultConfig.tooSoon, "too-soon", 76, "If NextUpdate is fewer than this many hours in future, warn.")
+	flag.BoolVar(&DefaultConfig.ignoreExpiredCerts, "ignore-expired-certs", false, "If a cert is expired, don't bother requesting OCSP.")
+	flag.IntVar(&DefaultConfig.expectStatus, "expect-status", -1, "Expect response to have this numeric status (0=Good, 1=Revoked, 2=Unknown)")
+	flag.IntVar(&DefaultConfig.expectReason, "expect-reason", -1, "Expect response to have this numeric revocation reason (0=Unspecified, 1=KeyCompromise, etc.)")
+	// Note that this init() does *not* call flag.Parse(). If a binary imports
```

Comment:
Given the caveats of using `flag.FooVar` mentioned in this comment, wouldn't it be simpler/less error-prone to instead keep the flags defined as globals like they were before, but replace `DefaultConfig` for interactive use-cases with a function `ConfigFromFlags()` which fetches the values of each flag into a new copy of the struct? This would also avoid polluting unit/integration tests with flag-specified behavior (they'd keep using the constant `DefaultConfig`).

Reply:
Yeah, I considered this approach, but initially didn't go with it because I didn't trust myself to have a full understanding of the behavior of all clients of this library. Looking at it this morning, this seems like a reasonable approach, so I've changed it to work this way.